### PR TITLE
#171380734 Configure ESLint

### PR DESCRIPTION
**Pull request by since9629**
#### What does this PR do?
Adds ESLint as a style guide linter to the project
#### Description of Task to be completed?
- create a .eslintrc.json file to hold configurations and AirBnB style guide rules
- Install `eslint ` npm package for linting
- Install `eslint-config-airbnb` npm package for AirBnB base configurations(Some rules are overridden in eslintrc.json file)
- Install `eslint-plugin-import` npm package to support linting of ES2015+ (ES6+) import/export syntax, and prevent issues with misspelling of file paths and import names
#### How should this be manually tested?
- After cloning the project repo, run `npm i` inside the cloned repo folder
- run `git pull <remote name>(e.g:origin) ch-eslint-configuration-171380734`
- If you are using vs-code text editor, [install ESLint plugin for vs-code](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
- configure eslint as the default formatter for your editor 
- create a JavaScript file inside the project folder you just cloned(preferably src folder) and add some JavaScript code, you can use `var invalid_var  =  0;` for testing
- If all of the above steps have been completed properly, vs-code should display some error messages
- In case there are some errors, run `./node_modules/.bin/eslint --fix <path to file||folder with errors>` to fix them
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#171380734
#### Screenshots (if appropriate)
N/A

### CODE REVIEW

**Comment by Muhire-Josue** [comment]
Great job Patrick, it looks well to me.

File: package.json [hunk 1]
```diff
@@ -26,5 +26,10 @@
     "passport-local": "^1.0.0",
     "request": "^2.87.0",
     "underscore": "^1.9.1"
```
Comment (ricahcyuzuzo) [inline package.json:32]: Hey Patrick, what are these for?

Comment (since9629) [reply package.json:32]: - passport-local is for authentication
- request is for creating http requests
- underscore.js according to the [docs](https://www.npmjs.com/package/underscore), is a utility-belt library for JavaScript that provides support for the usual functional suspects (each, map, reduce, filter...) without extending any core JavaScript objects.

**Review by ricahcyuzuzo — APPROVED** [review]

**Comment by since9629** [comment]
> Great job Patrick, it looks well to me.

great to hear that Josue, would you approve the PR then?

**Review by descholar-ceo — APPROVED** [review]
LGTM

**Review by Muhire-Josue — APPROVED** [review]
Good job

File: .gitignore [hunk 2]
```diff
@@ -10,7 +10,7 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 lerna-debug.log*
-
+package-lock.json
```
Comment (mifeille) [inline .gitignore]: Good job, but based on NPM documentation It is highly recommended you commit the generated package-lock.json file.
read this: https://docs.npmjs.com/configuring-npm/package-locks.html

**Review by muchristian — APPROVED** [review]

**Review by mifeille — APPROVED** [review]

**Pull request merged**